### DISCUSSION TITLE
Unable to mount additional volumes

### DISCUSSION THREAD

**Discussion by stevenhorner** [photoprism/photoprism#2246] [Q&A]
I've been struggling to add an additional volume to my Photoprism install. I receive the following error when I start photoprism with Docker Compose:

```
Error response from daemon: failed to create shim: OCI runtime create failed: container_linux.go:380: starting container process caused: process_linux.go:545: container init caused: rootfs_linux.go:75: mounting "/media/goliath" to rootfs at "/photoprism/originals/goliath" caused: mkdir /var/lib/docker/overlay2/b57b5bd2caedc1272e0ac8bbd49aca056eab90795a8d954e5f1a7132861dda0c/merged/photoprism/originals/goliath: permission denied: unknown
steven@docker:~$
```

Docker appears to be complaining about a permission error, but I can't see anything wrong with the permissions on the 2 volumes I have added to Photoprism. My Volumes are listed below as configured in docker-compose.yml

```
    volumes:
      # "/host/folder:/photoprism/folder"                # example
      - "/media/NextcloudSteven:/photoprism/originals"               # original media files (photos and videos)
      - "/media/goliath:/photoprism/originals/goliath" # older photos
```

If I only mount the originals (first volume listed) and comment out the goliath volume Photoprism works fine. I can browse the goliath folder though from my host and both volumes have the same permission, so don't think it's a problem with access the actual volumes. Both volumes  reside on other machines, the originals is on my nextcloud server which as mentioned above works with no issues if it's the only volume listed in the yml file. Goliath is on a Linux box with a NTFS volume mounted, I don't think this is the issue as I can see all files on the volume from the Docker host.

Am I missing something obvious here?

Thanks for any help, Photoprism is looking like a serious replacement for Google Photos if I can resolve adding additional volumes.

**Comment by lastzero** [comment]
It seems that you have problems with the file system permissions. Are you absolutely sure that the user you are running Docker/Docker Compose with has write permissions to all volumes and folders? Does it work as root / with the `sudo` command as described in our docs? In any case, it would be helpful to share the output of `ls -alh` to reduce the guesswork:

https://docs.photoprism.app/getting-started/troubleshooting/docker/#file-permissions

**Comment by stevenhorner** [comment]
Thanks for helping. I get the same error whether I run as my user or using sudo

```
steven@docker:~$ sudo docker-compose up
Removing steven_photoprism_1
steven_mariadb_1 is up-to-date
Recreating 59c0769007d5_steven_photoprism_1 ... error

ERROR: for 59c0769007d5_steven_photoprism_1  Cannot start service photoprism: fa                                                                                                                                                             iled to create shim: OCI runtime create failed: container_linux.go:380: starting                                                                                                                                                              container process caused: process_linux.go:545: container init caused: rootfs_l                                                                                                                                                             inux.go:75: mounting "/media/goliath" to rootfs at "/photoprism/originals/goliat                                                                                                                                                             h" caused: mkdir /var/lib/docker/overlay2/75268b14f0606a4aa451e8ce38b570d69e08e5                                                                                                                                                             7e09c549c5c9bb355782f0b38e/merged/photoprism/originals/goliath: permission denie                                                                                                                                                             d: unknown

ERROR: for photoprism  Cannot start service photoprism: failed to create shim: O                                                                                                                                                             CI runtime create failed: container_linux.go:380: starting container process cau                                                                                                                                                             sed: process_linux.go:545: container init caused: rootfs_linux.go:75: mounting "                                                                                                                                                             /media/goliath" to rootfs at "/photoprism/originals/goliath" caused: mkdir /var/                                                                                                                                                             lib/docker/overlay2/75268b14f0606a4aa451e8ce38b570d69e08e57e09c549c5c9bb355782f0                                                                                                                                                             b38e/merged/photoprism/originals/goliath: permission denied: unknown
ERROR: Encountered errors while bringing up the project.
steven@docker:~$ sudo docker-compose up
Removing steven_photoprism_1
steven_mariadb_1 is up-to-date
Recreating 59c0769007d5_steven_photoprism_1 ... error

ERROR: for 59c0769007d5_steven_photoprism_1  Cannot start service photoprism: failed to create shim: OCI runtime create failed: container_linux.go:380: starting container process caused: process_linux.go:545: container init caused: rootfs_linux.go:75: mounting "/media/goliath" to rootfs at "/photoprism/originals/goliath" caused: mkdir /var/lib/docker/overlay2/b2e48303a2ef537717a4b70c2cc1262155eaef9eae227815f68a20b7596cafd8/merged/photoprism/originals/goliath: permission denied: unknown

ERROR: for photoprism  Cannot start service photoprism: failed to create shim: OCI runtime create failed: container_linux.go:380: starting container process caused: process_linux.go:545: container init caused: rootfs_linux.go:75: mounting "/media/goliath" to rootfs at "/photoprism/originals/goliath" caused: mkdir /var/lib/docker/overlay2/b2e48303a2ef537717a4b70c2cc1262155eaef9eae227815f68a20b7596cafd8/merged/photoprism/originals/goliath: permission denied: unknown
ERROR: Encountered errors while bringing up the project.
steven@docker:~$
```

The permissions on the folders I am trying to mount.

```
steven@docker:~$ cd /media
steven@docker:/media$ ls -alh
total 8.0K
drwxr-xr-x  6 root root 4.0K Apr 12 14:47 .
drwxr-xr-x 20 root root 4.0K Mar 24 19:47 ..
drwxr-xr-x  2 root root    0 Aug  4  2017 1
drwxr-xr-x  2 root root    0 Mar 21  2018 cctv
drwxr-xr-x  2 root root    0 Mar 18 15:30 goliath
drwxr-xr-x  2 root root    0 Sep 21  2020 NextcloudSteven
```

**However, I decided to test setting the goliath volume as the the main originals and it worked, I then added my nextcloud folder as a second volume and it worked. I have no idea why it worked when in the opposite order.** 

This is a test install and feel I should redo it from scratch to ensure I haven't messed up the Docker install somehow.

**Reply by lastzero** [reply]
Have you already googled for this error? It doesn't seem to be specific to PhotoPrism, but rather a general Docker issue. You could also try mounting folders side by side.

Note, however, that files/folders added to the root directory may get lost this way after a container restart, e.g. if you use the web upload, since we don't provide a folder selector for that yet. A workaround is to add an anonymous `/photoprism/originals` volume as in this example:

```yaml
    volumes:
      - "/photoprism/originals"
      - "/media/NextcloudSteven:/photoprism/originals/nextcloud"
      - "/media/goliath:/photoprism/originals/goliath"
```

**Reply by stevenhorner** [reply]
Thanks, I did do a Google search but none of the results appeared to directly relate to my problem.

Like I mentioned switching the order around and it magically worked, no idea why it made a difference. I have just managed to fill the 50GB test VM with cached thumbnails. So will ensure I setup as per your volume example above and look at a better location for Photoprisms Storage folder.

Thanks again for the excellent software, once I have it up running properly I will look at Github Sponsorship or Patreon.

**Comment by heisenberg2980** [comment]
@lastzero Same issue here, Photoprism running using docker-compose without any issue having just one folder (/mnt/Photoprism_originals), but when I try to add a second folder with the same permissions (/mnt/Photoprism_originals_video), I get this error:

```
Error response from daemon: failed to create shim: OCI runtime create failed: container_linux.go:380: starting container process caused: process_linux.go:545: container init caused: rootfs_linux.go:75: mounting "/mnt/Photoprism_originals_video" to rootfs at "/photoprism/originals/video" caused: mkdir /var/lib/docker/overlay2/64be6cbcbfdfc4ab96c7ef27a14cb81d7b8630224fb0ddfd791ff4667b9b5eba/merged/photoprism/originals/video: read-only file system: unknown
```

This is the config in docker-compose.yml:

```
    working_dir: "/photoprism"
    ## Storage Folders: "~" is a shortcut for your home directory, "." for the current directory
    volumes:
      # "/host/folder:/photoprism/folder"                # example
      #- "/mnt/Photoprism/Photos:/photoprism/originals"               # original media files (photos and videos)
      - "/mnt/Photoprism_originals:/photoprism/originals"               # original media files (photos and videos)
      - "/mnt/Photoprism_originals_video:/photoprism/originals/video"
      # - "/example/family:/photoprism/originals/family" # *additional* media folders can be mounted like this
      #- "/mnt/Photoprism/Photos/Import:/photoprism/import"                  # *optional* base folder from which files can be imported to originals
      - "/mnt/Photoprism/storage:/photoprism/storage"                  # *writable* storage folder for cache, database, and sidecar files (never remove)
```

Permissions of both folders are the same:

```
root@docker ~/docker-compose/photoprism# ls -lha /mnt/
total 20K
drwxr-xr-x  5 root root    4.0K Jul  8 17:30 .
drwxr-xr-x 18 root root    4.0K Jul  8 18:06 ..
drwxr-x--x  3  110 nogroup 4.0K Jun  1 18:09 Photoprism
drwxr-xr-x 22  110    1000 4.0K May 10 21:00 Photoprism_originals
drwxr-xr-x  3  110    1000 4.0K Jul  8 18:08 Photoprism_originals_video
```

I tried @stevenhorner solution of switching the folders but it didn´t work either, any other idea?

**Reply by lastzero** [reply]
Seems to be a local permissions issue, maybe the current users doesn't not have enough permissions to create a new volume?

**Reply by heisenberg2980** [reply]
I am running the "docker-compose up" command with root user. Do you mean the user might not have enough permissions to create the folder "/photoprism/originals/video" inside the container? how can I check that?

**Reply by heisenberg2980** [reply]
I have just done an interesting test, I renamed the name of the container folder to "/photoprism/**originals2**/video" and the container started without any issue, so the problem is the container is not able to create a folder inside the originals folder (I also tried changing "video" for a different name, just in case the issue was related to the subfolder name, but the result is the same error).

This is the line of docker-compose.yml I changed for this test:

```
- "/mnt/Photoprism_originals_video:/photoprism/originals2/video"
```

I guess the folder needs to be created inside of the originals folder for photoprism to pick it up and scan it. Any idea why this could be happening?

**Comment by heisenberg2980** [comment]
I just found the problem (and the solution).

For anyone having this issue, if you want to add a new folder to index, you have to change the first folder (/photoprism/originals) to a subfolder (i.e. "/photoprism/originals/photo"), so the second folder you are adding is NOT a subfolder of the first one.

The only problem when you use this solution is that the first folder then has a different path so it needs to be re-indexed, which is my case is not a good option as the full indexing process takes around a week. @lastzero any idea to avoid the full re-index when changing the original folder path from /photoprism/originals to /photoprism/originals/photo? 

It would also be a good idea to replace this folder to a subfolder in the docker-compose.yml examples so other people don´t have this issue in the future.

**Comment by Kafkalasch** [comment]
I just had the same issue. 
Wondered how I lost almost 4000 photos as the indexing thought that all photos were deleted.

Looked into the container and realized that all subfolders were not mounted at all. But I did not get any error messages. (or at least could not find them)

stopping, updating and restarting the container did remount the folders again.
now started the indexing process (`photoprism index -a`). Hope this will fix the missing photos.

Would be interesting why it did work before and now again but stopped working inbetween. If it happens again, I will try @Heisenberg2980 's solution

**Comment by heisenberg2980** [comment]
@stevenhorner can you check my solution and if that solves your issue mark this question as solved?

@lastzero not sure if this is happening to everyone or just some people (maybe you or someone else can test it), but I think the solution I proposed in my previous comment is pretty easy (i.e. changing "/photoprism/originals" to "/photoprism/originals/main" in the file https://dl.photoprism.app/docker/docker-compose.yml), could you do this change? happy to do a pull request if you point me to where this file is in github
